### PR TITLE
WIP BXMSDOC-8003: Added Metadata attribute property and reuse data types feature as per BAPL-1936 and BAPL-1639

### PR DESCRIPTION
**Pull request by hmanwani-rh**
See the following JIRAs: 
https://issues.redhat.com/browse/BXMSDOC-8003
https://issues.redhat.com/browse/BXMSDOC-8004

See the following BAPLs: 
https://issues.redhat.com/browse/BAPL-1936
https://issues.redhat.com/browse/BAPL-1639

Doc preview:
[Designing business processes using BPMN models](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-8003-RHPAM/#con_custom-tasks-overview)

### CODE REVIEW

**Review by Josephblt — APPROVED** [review]

**Review by Josephblt — APPROVED** [review]

File: doc-content/enterprise-only/bpmn/proc-create-process-data-object.adoc [hunk 1]
```diff
@@ -23,7 +23,12 @@ In {PRODUCT} {PRODUCT_VERSION_LONG}, limited support for data objects is provide
 |Description
 
 | *Name*
-| The name of the data object. You can also double-click the data object shape to edit the name. 
+| The name of the data object. You can also double-click the data object shape to edit the name.
+
+| *Metadata Attributes*
+a| Add any custom metadata attribute name and value that you want to use for custom event listeners, such as a listener to implement some action when a metadata attribute is present.
+
+The *Metadata Attributes* enable the new `metaData` extensions to BPMN diagrams and modify the behavior of the overall data object.
```
Comment (mramendi) [inline doc-content/enterprise-only/bpmn/proc-create-process-data-object.adoc:31]: "modify the behavior of the overall data object" seems unclear to me. Do these attributes matter only with custom listeners? Or do they actually affect "behaviour of the object"? If the latter, then in what way do they affect this behaviour?

(Will not repeat this note for the same phrase elsewhere)

File: doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc [hunk 2]
```diff
@@ -78,6 +78,11 @@ In the BPMN modeler, you can configure the following general properties for a se
 |*Assignments*
 |Defines data input and output for the task.
 
+| *Metadata Attributes*
+a| Defines the custom metadata attribute name and value that you want to use for custom event listeners, such as a listener to implement some action when a metadata attribute is present.
+
+The *Metadata Attributes* enable the new `metaData` extensions to BPMN diagrams and modify the behavior of the overall task.
```
Comment (mramendi) [inline doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc:84]: Again "modify the behaviour" is unclear to me.

File: doc-content/enterprise-only/bpmn/proc-create-process-data-object.adoc (hunk 1 above)
Comment (hmanwani-rh) [reply doc-content/enterprise-only/bpmn/proc-create-process-data-object.adoc:31]: @mramendi Let's say the following keys and values are added to the metadata attribute for a task:
image

Then using metadata attribute, it is identified whether the task's attribute is defined or it is needed to be used in conjunction with a custom event listener.

File: doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc (hunk 2 above)
Comment (hmanwani-rh) [reply doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc:84]: Please see the above explanation

File: doc-content/enterprise-only/bpmn/bpmn-con.adoc [hunk 3]
```diff
@@ -36,7 +36,7 @@ A process is defined or determined by its process definition. It exists in a kno
 | Enter a description of the purpose of the process.
 
 | *Imports*
-| Click to open the *Imports* window and add any data type classes required for your process.
+| Click to open the *Imports* window and add any data type classes required for your process, or you can select an already defined data type.
```
Comment (briandooley) [inline doc-content/enterprise-only/bpmn/bpmn-con.adoc:39]:
```suggestion
| Click to open the *Imports* window.
| Add any data type classes your process requires, or you can select an already defined data type.
```

File: doc-content/jbpm-docs/src/main/asciidoc/BPMN2/proc-create-process-data-object.adoc [hunk 4]
```diff
@@ -23,10 +23,17 @@ In {PRODUCT} {PRODUCT_VERSION_LONG}, limited support for data objects is provide
...
+| *Metadata Attributes*
+a| Add any custom metadata attribute name and value that you want to use for custom event listeners, such as a listener to implement some action when a metadata attribute is present.
+
+The *Metadata Attributes* enable the new `metaData` extensions to BPMN diagrams and modify the behavior of the overall data object.
 
 | *Type*
-| Select a type of the data object.
+a| Select a type of the data object.
+
+IMPORTANT: When you define a data type for a data object, you can use the same data type for other data type fields in *Properties*, such as *Imports* and *Data Assignments*.
```
Comment (briandooley) [inline doc-content/jbpm-docs/src/main/asciidoc/BPMN2/proc-create-process-data-object.adoc:36]:
```suggestion
IMPORTANT: When defining a data type for a data object, you can use the same data type for other data type fields in *Properties*, such as *Imports* and *Data Assignments*.
```

**Review by briandooley — CHANGES_REQUESTED** [review]
Reviewed [this commit](https://github.com/hmanwani-rh/kie-docs/pull/66/commits/52ce5a5507bb22c4f6219d5c9b802e26d1257cb8). Minor changes suggested.

**Review by romartin — APPROVED** [review]

File: doc-content/enterprise-only/bpmn/bpmn-con.adoc (hunk 3 above)
Comment (hmanwani-rh) [reply doc-content/enterprise-only/bpmn/bpmn-con.adoc:39]: The above statement is available in a table, that's why we can't split it.

**Review by MarianMacik — APPROVED** [review]
Looks good, thanks!

**Pull request merged**